### PR TITLE
dtoh: Refactor ignored... comments into a function

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
Removes some really ugly duplication and makes the code more concise

### CODE REVIEW

File: src/dmd/dtoh.d [hunk 1]
```diff
@@ -422,11 +422,7 @@ public:
 
         if (isBuildingCompiler && s.getModule() && s.getModule().isFrontendModule())
         {
-            if (printIgnored)
-            {
-                buf.printf("// ignored %s %s", s.kind(), s.toPrettyChars());
-                buf.writenl();
-            }
+            ignore("// ignored %s %s", s.kind(), s.toPrettyChars());
```
Comment (thewilsonator) [inline src/dmd/dtoh.d:425]: if you're refactoring you may as well refactor the leading `"// ignored"`

File: src/dmd/dtoh.d [hunk 2]
```diff
@@ -1983,4 +1931,37 @@ public:
         }
         buf.writeByte(')');
     }
+
+    static if (__VERSION__ < 2092)
+    {
+        private void ignore(const char* format, ...) nothrow
+        {
+            import core.stdc.stdarg;
+            if (printIgnored)
```
Comment (thewilsonator) [inline src/dmd/dtoh.d:1940]:
```
if (!printIgnored)
     return;
```

File: src/dmd/dtoh.d (hunk 1 above)
Comment (MoonlightSentinel) [reply src/dmd/dtoh.d]: Good idea, done.

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
Need to update `compilable/test21217.d`

**Comment by MoonlightSentinel** [comment]
> Need to update `compilable/test21217.d`

Done

**Pull request merged**